specs/language/expressions.tex

```diff
@@ -160,3 +160,87 @@
 overloaded implementation of \texttt{operator[]} (\ref{Overload}).\footnote{HLSL
 does not support the base address of a subscript operator being the expression
 inside the braces, which is valid in C and C++.}
+
+\Sec{Function Calls}{Expr.Post.Call}
+
+\p A function call may be an \textit{ordinary function}, or a \textit{member
+function}. In a function call to an \textit{ordinary function}, the
+\textit{postfix-expression} must be an lvalue that refers to a function. In a
+function call to a \textit{member function}, the \textit{postfix-expression}
+will be an implicit or explicit class member access whose \textit{id-expression}
+is a member function name.
+
+\p When a function is called, each parameter shall be initialized with its
+corresponding argument. The order in which parameters are initialized is
+unspecified. \footnote{Today in DXC targeting DXIL matches the Microsoft C++ ABI
+and evaluates argument expressions right-to-left, while SPIR-V generation
+matches the Itanium ABI evaluating parameters left-to-right. There are good
+arguments for unifying these behaviors, and arguments for keeping them
+different.}
+
+\p If the function is a non-static member function the \texttt{this} argument
+shall be initialized to a reference to the object of the call as if casted by an
+explicit cast expression to an lvalue reference of the type that the function is
+declared as a member of.
+
+\p Parameters are either \textit{input parameters}, \textit{output parameters},
+or \textit{input/output parameters} as denoted in the called function's
+declaration (\ref{Decl.Function}).
+
```

Comment:
HLSL Function Arguments [description](https://learn.microsoft.com/en-us/windows/win32/direct3dhlsl/dx-graphics-hlsl-function-parameters) lists `uniform` in addition to the above as parameter annotations. Is this spec a forward-looking one with an intent to deprecate it (#135) and hence it was not listed?

Reply:
Found out that DXC does not support uniform parameters.